eureka-integration/src/main/java/com/netflix/eureka2/integration/WriteClusterIntegrationTest.java

```diff
@@ -76,19 +81,26 @@ public void testWriteClusterReplicationWithRegistrationLifecycle() throws Except
                 seedBuilder.withAppGroup("CCC").build()
         );
 
-        registrationClient.register(infos.get(0)).subscribe();
-        registrationClient.register(infos.get(1)).subscribe();
-        registrationClient.register(infos.get(2)).subscribe();
-        registrationClient.unregister(infos.get(2)).subscribe();
-
         // Subscribe to second write server
-        Iterator<ChangeNotification<InstanceInfo>> notificationIterator =
-                iteratorFrom(10, TimeUnit.SECONDS, discoveryClient.forApplication(infos.get(0).getApp()));
-
-        assertThat(notificationIterator.next(), is(addChangeNotificationOf(infos.get(0))));
-        assertThat(notificationIterator.next(), is(modifyChangeNotificationOf(infos.get(1))));
-        assertThat(notificationIterator.next(), is(modifyChangeNotificationOf(infos.get(2))));
-        assertThat(notificationIterator.next(), is(deleteChangeNotificationOf(infos.get(2))));
+        final BlockingQueue<ChangeNotification<InstanceInfo>> notifications = new LinkedBlockingQueue<>();
+        discoveryClient.forApplication(infos.get(0).getApp()).subscribe(new Action1<ChangeNotification<InstanceInfo>>() {
+            @Override
+            public void call(ChangeNotification<InstanceInfo> notification) {
+                notifications.add(notification);
+            }
+        });
+
+        // We need to block, otherwise if we shot all of them in one row, they will be processed
+        // in order, but subscribe operations may eventually re-ordered them.
```

Comment:
With the latest changes the subscribe operator should not reorder them if they are declared in order.